app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -251,6 +251,16 @@ public void init() {
         compositeDisposable.add(
                 eventCaptureRepository.eventSections()
                         .flatMap(sectionList -> fieldFlowable
+                                .map(fields->{
+                                    Iterator<FieldViewModel> iterator = fields.iterator();
+                                    while (iterator.hasNext()){
+                                        FieldViewModel field = iterator.next();
+                                        if(field instanceof DisplayViewModel){
+                                            iterator.remove();
+                                        }
+                                    }
+                                    return fields;
+                                })
```

Comment:
why not avoid adding these fields instead of removing them? it's not very clear why in this part of the code we are removing them

Reply:
Cause all these fields are needed for the forms. The source is the same for both showing the form fields and calculating the percentage. But for the later, you have to remove the display fields created by program rules.